include/core/physical_property_model.h

```diff
@@ -36,15 +36,29 @@ enum field : int
   previous_temperature
 };
 
+inline void
```

Comment:
Inlining getters is generally a good idea. However, if the function has inheritance (e..g class B derives from class A) inlining will not occur because of the need to go through a vtable. So the TLDR would be : If the function is used by the class or the class has no inheritance tree, inline functions in the .h if they are short, otherwise dont :).